### PR TITLE
Refactor context-menu options in table header

### PR DESCRIPTION
**Pull request by wolmir**
This is a follow-up of #2593 which is the target branch for this

It will extend the sorting options to some column placeholders that have a single child column.
The objective is to prevent an empty placeholder to have different options than the column whose place it is holding.

This will also move the header menu content to its own component in an effort to simplify the context-menu logic a bit.

### CODE REVIEW

**Comment by sroy3** [comment]
The actions do not work it triggered from the placeholder:

https://user-images.githubusercontent.com/3683420/197207133-83a54bc5-eb71-432a-b642-84c0cb59a0a8.mov

**Comment by wolmir** [comment]
> The actions do not work it triggered from the placeholder

Thanks! I'll try to add a test for the placeholder case

File: extension/src/experiments/columns/model.test.ts [hunk 1]
```diff
@@ -249,4 +249,62 @@ describe('ColumnsModel', () => {
...
+      await model.transformAndSet(outputFixture)
+      expect(model.getSelected()).not.toStrictEqual(columnsFixture)
+
+      model.toggleStatus(parentPath)
+
+      const children = model.getChildren(parentPath)
+      expectStatus(children, Status.SELECTED)
+    })
+
+    it('should unselect the toggled children of an unselected column', async () => {
```
Comment (wolmir) [inline extension/src/experiments/columns/model.test.ts:294]: This is the failing test for https://github.com/iterative/vscode-dvc/pull/2593#discussion_r1001869678

It took some time to find the issue. I will start fixing them today

**Comment by sroy3** [comment]
> The actions do not work it triggered from the placeholder:
> 
>  Screen.Recording.2022-10-21.at.9.27.23.AM.mov

I'm getting the same thing. Is that normal?

**Comment by wolmir** [comment]
> > The actions do not work it triggered from the placeholder:
> > Screen.Recording.2022-10-21.at.9.27.23.AM.mov
> 
> I'm getting the same thing. Is that normal?

No, not normal, it's sending the placeholder's synthetic id instead of the actual column id as the message payload. I'll fix it, thanks!

**Comment by wolmir** [comment]
@sroy3 @julieg18 Can you take another look here please?

**Review by julieg18 — COMMENTED** [review]
Looks like `yarn lint` is failing. Can you correct the problem?

**Comment by sroy3** [comment]
> > > The actions do not work it triggered from the placeholder:
> > > Screen.Recording.2022-10-21.at.9.27.23.AM.mov
> > 
> > 
> > I'm getting the same thing. Is that normal?
> 
> No, not normal, it's sending the placeholder's synthetic id instead of the actual column id as the message payload. I'll fix it, thanks!

It's still the same.

**Review by sroy3 — CHANGES_REQUESTED** [review]

**Comment by sroy3** [comment]
Fixed sorting on placeholders


https://user-images.githubusercontent.com/3683420/201367956-2bfca67b-b095-4326-a69e-3226e3478d16.mov

What's missing is hiding only columns underneath a parent, but I think we should do this as a follow-up

https://user-images.githubusercontent.com/3683420/201368426-c3527166-d799-4441-ac1e-8343bed82341.mov

**Review by sroy3 — APPROVED** [review]

**Pull request merged**